### PR TITLE
Fix-issues-related-to-playbooks

### PR DESCRIPTION
**Pull request by mainawycliffe**

### CODE REVIEW

File: src/api/services/playbooks.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export async function getPlaybookToRunForResource(
 export async function getPlaybookRun(id: string) {
   const res = await IncidentCommander.get<PlaybookRunWithActions[] | null>(
     // todo: use playbook names instead
-    `/playbook_runs?id=eq.${id}&select=*,created_by(${AVATAR_INFO}),playbooks(id,name,spec),component:components(id,name,icon),actions:playbook_run_actions(id,name,status,start_time,end_time)`
+    `/playbook_runs?id=eq.${id}&select=*,created_by(${AVATAR_INFO}),playbooks(id,name,spec),component:components(id,name,icon),actions:playbook_run_actions(id,name,status,start_time,end_time,schedule_time)`
```
Comment (adityathebe) [inline src/api/services/playbooks.ts:87]: This would be `scheduled_time`

File: src/api/types/playbooks.ts [hunk 2]
```diff
@@ -58,7 +66,22 @@ export type PlaybookSpec = {
   icon?: string;
   created_by?: User;
   category?: string;
-  spec: any;
+  spec: {
+    actions?: Record<string, any>[];
+    configs?: {
+      type?: string;
```
Comment (adityathebe) [inline src/api/types/playbooks.ts:72]: We also need tags in here

https://github.com/flanksource/mission-control/blob/c1b6368a9e1ebd010c8ab32965bfdf6320e74f0c/api/v1/playbook_types.go#L18-L23

File: src/components/Playbooks/Runs/Actions/PlaybookRunsActionItem.tsx [hunk 3]
```diff
@@ -34,10 +35,23 @@ export default function PlaybookRunsActionItem({
...
+      <div className="flex flex-row gap-2">
         <div className="text-sm  text-gray-600">
           <Age from={action.start_time} to={action.end_time} />
         </div>
+        {/* only show, if scheduled time is in the future */}
+        {action.status === "sleeping" &&
+          action.scheduled_time &&
+          dayjs().isBefore(dayjs(action.scheduled_time)) && (
+            <div className="text-sm text-gray-600">
+              Resumes in
```
Comment (adityathebe) [inline src/components/Playbooks/Runs/Actions/PlaybookRunsActionItem.tsx:47]: Seems to be a typo here

image

File: src/api/services/playbooks.ts (hunk 1 above)
Comment (mainawycliffe) [reply src/api/services/playbooks.ts:87]: Wierd, that should have thrown an error and it didn't when I was testing let me take a look.

**Comment by adityathebe** [comment]
image

I think the durations in here needs to have labels _(or any other way to make it more obvious what those times are)_. Cuz here I'm seeing 3 different durations
- execution completion duration (in 5s)
- started duration (43s)
- resumption duration (in 7s)

**Comment by moshloop** [comment]
@adityathebe what is the started duration ?

**Comment by moshloop** [comment]
@mainawycliffe For sleeping actions don't show the start time/age, use a `-` and a tooltip for the resume time

**Comment by mainawycliffe** [comment]
@moshloop for currently running steps, should we show a loading animation?

**Pull request merged**